pkg/synopsysctl/cmd_migrate.go

```diff
@@ -298,6 +299,29 @@ func migrateAlert(namespace string) error {
 		alertNamespace := alert.Spec.Namespace
 		log.Infof("migrating Alert '%s' in namespace '%s'...", alertName, alertNamespace)
 
+		// default images
+		defaultAlertImages := []string{
+			"docker.io/blackducksoftware/blackduck-alert:3.1.0",
+			"docker.io/blackducksoftware/blackduck-alert:4.0.0",
+			"docker.io/blackducksoftware/blackduck-alert:4.1.0",
+			"docker.io/blackducksoftware/blackduck-alert:4.2.0",
+		}
+		defaultCfsslImage := "docker.io/blackducksoftware/blackduck-cfssl:1.0.0"
+
+		// get the current value of the images using reflect
+		alertImage := reflect.ValueOf(alert.Spec).Elem().FieldByName("AlertImage").Interface().(string)
+		cfsslImage := reflect.ValueOf(alert.Spec).Elem().FieldByName("CfsslImage").Interface().(string)
+
+		// check whether the cfssl image is not equal to default image, if so, update the image registry
+		if len(cfsslImage) > 0 && !strings.EqualFold(cfsslImage, defaultCfsslImage) && !util.IsExistInStringSlice(alert.Spec.ImageRegistries, cfsslImage) {
+			alert.Spec.ImageRegistries = append(alert.Spec.ImageRegistries, cfsslImage)
+		}
+
+		// check whether the alert image is not equal to default image, if so, update the image registry
+		if len(alertImage) > 0 && util.IsExistInStringSlice(defaultAlertImages, alertImage) && !util.IsExistInStringSlice(alert.Spec.ImageRegistries, alertImage) {
```

Comment:
shouldn't this be !util.IsExistInStringSlice(defaultAlertImages, alertImage)?

Reply:
fixed